h2o-py/h2o/explanation/_explain.py

```diff
@@ -1305,35 +1308,59 @@ def ice_plot(
             factor_map = _factor_mapper(NumpyFrame(frame[column]).from_factor_to_num(column))
 
         deciles = [int(round(frame.nrow * dec / 10)) for dec in range(11)]
+        deciles[10] = frame.nrow - 1
         colors = plt.get_cmap(colormap, 11)(list(range(11)))
         plt.figure(figsize=figsize)
         for i, index in enumerate(deciles):
+            percentile_string = "{}th Percentile".format(i * 10)
             tmp = NumpyFrame(
                 model.partial_plot(
                     frame,
                     cols=[column],
                     plot=False,
                     row_index=index,
                     targets=target,
-                    nbins=20 if not is_factor else 1 + frame[column].nlevels()[0]
+                    nbins=100 if not is_factor else 1 + frame[column].nlevels()[0],
+                    include_na=True
                 )[0]
             )
+            user_splits = dict()
+            orig_value = frame.as_data_frame()[column][index]
             encoded_col = tmp.columns[0]
+            if _isnan(orig_value):
+                id = np.where(tmp[encoded_col] == tmp.from_factor_to_num(encoded_col)['.missing(NA)'])[0][0] if is_factor else np.where(np.isnan(tmp[encoded_col]))[0][0]
+                orig_null_value = tmp.from_num_to_factor(encoded_col)[tmp[encoded_col][id]] if is_factor else tmp[encoded_col][id]
+                interval = "[{}, {}]".format(orig_null_value,  tmp["mean_response"][id])
+                warnings.warn("Original observation of \"" + encoded_col + "\" for " + percentile_string + " is " + interval + ". Ploting of NAs is not yet supported." )
```

Comment:
ultranit/style: This is just observation, but there are multiple string interpolation styles used (`.format` vs concatenating). Please consider unifying it. I'm not sure if the codebase needs to stay Python2 compatible, but elegant modern alternative is to use [Python F-strings](https://realpython.com/python-f-strings/)

Reply:
unified 👍  it has to stay python2 compatible 
[77c3d0a](https://github.com/h2oai/h2o-3/pull/6020/commits/77c3d0acca2ed68f6c134d67015734bc7c5af0f3)